src/Microsoft.ML.Data/TrainCatalog.cs

```diff
@@ -366,6 +368,99 @@ internal BinaryClassificationTrainers(BinaryClassificationCatalog catalog)
             return result.Select(x => new CrossValidationResult<CalibratedBinaryClassificationMetrics>(x.Model,
                 Evaluate(x.Scores, labelColumn), x.Scores, x.Fold)).ToArray();
         }
+
+        /// <summary>
+        /// The list of trainers for performing binary classification.
+        /// </summary>
+        public CalibratorsCatalog Calibrators { get; }
+
+        public sealed class CalibratorsCatalog : CatalogInstantiatorBase
+        {
+            internal CalibratorsCatalog(BinaryClassificationCatalog catalog)
+                : base(catalog)
+            {
+            }
+            /// <summary>
+            /// Adds probability column by training naive binning-based calbirator.
```

Comment:
>naive binning-based [](start = 52, length = 19)

can we link to any online resources? @yaeldekel, does this apply: [https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4410090/](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4410090/)

Reply:
I doubt they're the same. That article seems to have been published in 2015, whereas this calibrator has AFAIK existed at least since 2013, and probably goes back a fair bit further than that. I imagine Misha didn't just conjure up the method out of absolute thin air, but whatever work he based the implementation on has probably be lost to the sands of time.

---
In reply to: [261052618](https://github.com/dotnet/machinelearning/pull/2766#discussion_r261052618) [](ancestors = 261052618)